### PR TITLE
Add server and client specific path 

### PR DESCRIPTION
**Pull request by K1li4nL**
This pr introduces change to the path clients use to connect. 
With this pr, the following paths are available: 

`/server`: For servers to connect to
`/client`: For clients, this path replaces `/` we had until now. 

For now, client and servers are treated the same way no matter the path used to connect, this will change in the future.

### CODE REVIEW

File: be2-scala/README.md [hunk 1]
```diff
@@ -46,7 +46,7 @@ ch_epfl_pop_Server {
 	http {
 		interface = "127.0.0.1"
 		port = "8000"
-		path = ""
+		path = "client"
```
Comment (Jharaxus) [inline be2-scala/README.md:49]: The final update is not completely reflected: it misses `client-path = "client"` and `server-path = "server"`

File: be2-scala/src/main/scala/ch/epfl/pop/Server.scala [hunk 2]
```diff
@@ -46,16 +46,24 @@ object Server {
...
 
-      def publishSubscribeRoute: RequestContext => Future[RouteResult] = path(config.path) {
-        handleWebSocketMessages(PublishSubscribe.buildGraph(pubSubMediatorRef, dbActorRef, messageRegistry)(system))
+      // Setup routes
+      def publishSubscribeRoute: RequestContext => Future[RouteResult] = {
+        path(config.clientPath) {
+          handleWebSocketMessages(PublishSubscribe.buildGraph(pubSubMediatorRef, dbActorRef, messageRegistry)(system))
+        } ~ path(config.serverPath) {
+          handleWebSocketMessages(PublishSubscribe.buildGraph(pubSubMediatorRef, dbActorRef, messageRegistry)(system))
+        }
```
Comment (Jharaxus) [inline be2-scala/src/main/scala/ch/epfl/pop/Server.scala:55]: As both routes are the same they can be concatenated using `(path(config.clientPath) | path(config.serverPath)) {handleWebSocketMessages(...)}`

**Review by Jharaxus — CHANGES_REQUESTED** [review]
Two small nitpicks, otherwise good job!

File: be2-scala/src/main/scala/ch/epfl/pop/Server.scala (hunk 2 above)
Comment (K1li4nL) [reply be2-scala/src/main/scala/ch/epfl/pop/Server.scala:55]: Thats only temporary, the next steps are to change `buildGraph` so it behaves slightly differently based on the path used. More precisely: When `/server` is used, the clientActorRef will be sent to the `ConnectionMediator` (pr #1503 )
And the clientActor will also tell `ConnectionMediator` when it disconnects. It took me some time to figure out the `~`, I figured this way already have our template ready.

File: be2-scala/README.md (hunk 1 above)
Comment (K1li4nL) [reply be2-scala/README.md]: Good catch, done !

File: be2-scala/src/main/scala/ch/epfl/pop/Server.scala (hunk 2 above)
Comment (Jharaxus) [reply be2-scala/src/main/scala/ch/epfl/pop/Server.scala:55]: Ok that makes sense !

**Review by Jharaxus — APPROVED** [review]

**Review by Ajkunas — APPROVED** [review]
LGTM !

**Pull request merged**